lantern/numpy.py

```diff
@@ -1,34 +1,25 @@
 import numpy as np
-import io
-import cv2
+import torch
 
 
 class Numpy(np.ndarray):
-    @staticmethod
-    def from_matplotlib_figure(figure):
-        buffer = io.BytesIO()
-        figure.savefig(buffer, format="png", dpi=90, bbox_inches="tight")
-        buffer.seek(0)
-        image = np.frombuffer(buffer.getvalue(), dtype=np.uint8)
-        buffer.close()
-        image = cv2.imdecode(image, 1)
-        return image
-
     @classmethod
     def __get_validators__(cls):
         yield cls.validate
 
     @classmethod
-    def validate(cls, data):
+    def validate(cls, data) -> np.ndarray:
         if isinstance(data, cls):
             return data.view(np.ndarray)
         elif isinstance(data, np.ndarray):
             return data
+        elif isinstance(data, torch.Tensor):
+            return data.numpy()
         else:
             return np.array(data)
 
     @classmethod
-    def ndim(cls, ndim):
+    def ndim(cls, ndim) -> "Numpy":
```

Comment:
Since lantern is python 3.7+
```
from __future__ import annotations
```
and have it the "modern way" ` -> Numpy`

Reply:
```
NameError: name 'Numpy' is not defined
```

Reply:
Nevermind, you are right. It works with the import :+1: